docs/io/index_unofficial.rst

```diff
@@ -0,0 +1,283 @@
+found.
+
+The reader must also implement a fast method for
+`ctapipe.io.eventfilereader.EventFileReader.check_file_compatibility`. This
+method attempts to find a reader which *may* be compatible with the input
+file, and can be as simple as looking for a unique file extension in the file
+path or attempting to find a header keyword in the file. It is strongly
+advised that this method does not depend on the external reader software, as
+this would introduce a dependency when choosing the correct reader in the
+`ctapipe.io.eventfilereader.EventFileReaderFactory`.
+
+The currently existing EventFileReaders,
+`ctapipe.io.eventfilereader.HessioFileReader` and
+`ctapipe.io.unofficial.eventfilereader.TargetioFileReader` can be used as
+templates to create a new EventFileReader for a different data format.
+
+
+Low-Level Reader-Specific Modules
+===========================
+
+A seperate module should be created for the low-level method used for filling
+the event containers for a specific data format. The naming convention for
+this module is "\*io.py", where "\*" is the short-name for the data format.
+For example: `ctapipe.io.hessio` and `ctapipe.io.unofficial.targetio.targetio`.
+
+The low-level module can be a directory of multiple modules, allowing the
+various components involved in the event building (such as containers) to be
+grouped together for a particular data format.
+
+Each data format can be wildly different, therefore these low-level
+modules can appear very different in how they function. This is fine as long
+as these rules are followed:
+
+* The `ctapipe.io.eventfilereader.EventFileReader` calls the low-level methods
+  such that the high-level interface remains the same as other
+  `ctapipe.io.eventfilereader.EventFileReader`.
+* The low-level methods fill the containers with the relevant data, e.g. the
+  waveforms are stored in "data.r0.tel[].adc_samples" (assuming they are
+  waveforms of data level R0, see the next section for an explanation of data
+  levels and see `ctapipe.io.hessio` and
+  `ctapipe.io.unofficial.targetio.targetio` for examples)
+* The unique 'origin' string is added to the meta dictionary of the container
+  so the data source of the events can be obtained by other algorithms, e.g.
+
+  >>> data.meta['origin'] = "targetio"
+
+
+Data Levels (R0/R1/DL0/DL1)
+===========================
+
+It is important to read your events into the correct corresponding event
+container:
+
+* If the file contains raw events, that still require low-level calibration
+  to be useful for offline analysis, then they should be read into the r0
+  container.
+* If the file contains events which have their low-level calibration fully
+  applied to them, then they should be read into the r1 container.
+* If the file contains events which have their low-level calibration fully
+  applied to them AND they are in a proposed format of DL0 (perhaps including
+  a data reduction approach), then they should be stored in the dl0 container.
```

Comment:
"DL0" and "dl0" both used in the same sentence without making clear what the distinction is between capitalized and lower case version.

Reply:
Also this is a circular description. According to this sentence, a DL0 container contains data which is in a proposed DL0 format. 

Btw. "**a** proposed DL0 format", this implies there are multiple proposals and all are valid in this context.

Reply:
The lowercase refers to the name of the sub-container in the event container. I have given the lowercase code highlighting now.

It's not circular, but I am outlying the obvious just for clarity. If an unofficial data format is a proposed DL0 format (i.e. a format being proposed to CTA for storing the Data-Level-0 information to disk) then the information should be read into the `dl0` container.

Reply:
I will improve the explanation in the docs